feat: Add Agile InnerSource Dojo pattern (#695)
gruetter/innerSourceTest: patterns/2-structured/agile-innersource-dojo.md

## Problem

Agile projects, especially those using Scrum and promoting autonomy and self-organization, can unintentionally lead to silos.  Similar teams across the organization may create non-uniform and redundant codebases, resulting in inefficiencies and waste.  This can also lead to a compulsion to over-standardize practices, which stifles innovation, particularly in DevOps and other transformational initiatives.  Symptoms include low engagement in InnerSource projects, inefficient knowledge transfer and difficulty in scaling Agile practices.

## Context

A dojo is a didactic participatory practice group, predicated on traditional martial arts dojo practices. It creates opportunities for participants to learn by doing (embodied learning), under supervision and guidance of others who have prior experience. Agile InnerSource Dojos are designed for organizations aiming to harness the power of InnerSource to increase potentials for transformation and innovation.  This setting brings individuals from various backgrounds, expertise and roles to collaboratively work on shared goals, ensuring the learning journeys remain relevant, beneficial and peer supported.  

## Forces

> * **Lack of Awareness**: Many Agile practioners are unware of InnerSource methodologies and vice versa.

Reviewer: ```suggestion
* **Lack of Awareness**: Many Agile practitioners are unaware of InnerSource methodologies and vice versa.
```

* **Use Collaborative Tools**: Utilize tools like Jitsi Meet, Slack and GitHub for communication and documentation.  Insights can be optionally shared on social channels, ensuring privacy.
* **Encourage Voluntary Code Contribution**: Foster an environment where contributions are voluntary, allowing individuals to work on what interests them and aligning with their passions.
* **Prioritized Mentorship**: Implement a mentorship structure with Sensei (teachers) providing guidance and Senpai (senior students) supporting newer members. This dual-layer mentorship enhances skill development and community cohesion.
* **Foster a Culture of Continuous Improvement**: Regularly seek feedback to identify improvement areas, refining the Dojo's conditioning and activities accordingly.

## Resulting Context

Embracing this pattern helps InnerSource programs become vibrant, effective and sustainable.  Participants will feel more connected and engaged, leading to increased knowledge sharing, innovation and collective growth.

## Related Patterns

> * [Dedicated Community Leader](https://patterns.innersourcecommons.org/p/dedicated-community-leader)

Reviewer: ```suggestion
* [Dedicated Community Leader](../2-structured/dedicated-community-leader.md)
```

Better to use relative links to the markdown files of the other patterns. These will be turned into correct links in our book automatically.

When linking to files in folder 1-initial, these will then link to the files in our github repo, rather than in our online book.

So in other words, everything will work as intended out of the box once you use relative links here :)

Could you also change this for the others?

* [Core Team](https://patterns.innersourcecommons.org/p/core-team)
* [Trusted Committer](https://patterns.innersourcecommons.org/p/trusted-committer)
* [Maturity Model](https://patterns.innersourcecommons.org/p/maturity-model)
* [Circle Communities](https://patterns.innersourcecommons.org/p/circle-communities)

## Known Instances

* [InnerSource Dojo from SAP](https://www.youtube.com/watch?v=fXoVm5iTSCc)
  * [SAP Samples Dojo Model (live demo) - Mindset | Agile | InnerSource](https://sap-samples.github.io/dojo)
  * [SAP Samples Dojo Model (source) - Mindset | Agile | InnerSource](https://github.com/SAP-samples/dojo)
  * [SAP Samples GitHub Pull Analyzer](https://github.com/SAP-samples/github-pull-analyzer)
* [Microsoft DevOps Dojo](https://innersourcecommons.org/stories/microsoft)
* [Dojo Center](https://dojo.center)

>   * [Source](https://github.com/dojo-center/dojo-center.github.io) 

Reviewer: ```suggestion
  * [Source](https://github.com/dojo-center/dojo-center.github.io)
```
